examples/official-storybook/config.js

```diff
@@ -58,75 +58,6 @@ addParameters({
   ],
 });
 
-let previousExports = {};
-if (module && module.hot && module.hot.dispose) {
-  ({ previousExports = {} } = module.hot.data || {});
-
-  module.hot.dispose(data => {
-    // eslint-disable-next-line no-param-reassign
-    data.previousExports = previousExports;
-  });
-}
-
-// The simplest version of examples would just export this function for users to use
-function importAll(context) {
-  const storyStore = window.__STORYBOOK_CLIENT_API__._storyStore; // eslint-disable-line no-undef, no-underscore-dangle
-
-  context.keys().forEach(filename => {
-    const fileExports = context(filename);
-
-    // A old-style story file
-    if (!fileExports.default) {
-      return;
-    }
-
-    const { default: component, ...examples } = fileExports;
-    let componentOptions = component;
-    if (component.prototype && component.prototype.isReactComponent) {
-      componentOptions = { component };
-    }
-    const kindName = componentOptions.title || componentOptions.component.displayName;
-
-    if (previousExports[filename]) {
-      if (previousExports[filename] === fileExports) {
-        return;
-      }
-
-      // Otherwise clear this kind
-      storyStore.removeStoryKind(kindName);
-      storyStore.incrementRevision();
-    }
-
-    // We pass true here to avoid the warning about HMR. It's cool clientApi, we got this
-    const kind = storiesOf(kindName, true);
-
-    (componentOptions.decorators || []).forEach(decorator => {
-      kind.addDecorator(decorator);
-    });
-    if (componentOptions.parameters) {
-      kind.addParameters(componentOptions.parameters);
-    }
-
-    Object.keys(examples).forEach(key => {
-      const example = examples[key];
-      const { title = key, parameters } = example;
-      kind.add(title, example, parameters);
-    });
-
-    previousExports[filename] = fileExports;
-  });
-}
-
-function loadStories() {
-  let req;
-  req = require.context('../../lib/ui/src', true, /\.stories\.js$/);
-  importAll(req);
-
-  req = require.context('../../lib/components/src', true, /\.stories\.tsx?$/);
-  importAll(req);
-
-  req = require.context('./stories', true, /\.stories\.js$/);
-  importAll(req);
-}
-
-configure(loadStories, module);
+load(require.context('../../lib/ui/src', true, /\.stories\.js$/), module);
```

Comment:
AFAIK require.context needs these args in this way because it's implemented as some kind of macro. But I could be wrong. @tmeasday can you shed light here?

Reply:
If we have an api that takes the same arguments, we can do a bit of feature detection for users.

making storybook easier to consume from node:

```
loadGlobs('../../lib/ui/src', true, /\.stories\.js$/)
```

in jest will turn into a fs call to load and require all these files

in webpack will turn into require.resolve

that's the idea

Reply:
@ndelangen `require.context` is super finicky (or was last time I checked). If you don't call it with a constant string as first argument it will not work. i.e. `require.context("foo",...)` is OK, `const string="foo"; require.context(string, ...)` is not.

I think the best we could do is a babel plugin which is probably too much.